client/src/components/navBar/loginButton.tsx

```diff
@@ -1,7 +1,19 @@
 import { Button } from '@mui/material'
+import { useHistory } from 'react-router-dom'
+import { TokenManager } from '../../hooks/token'
 
-export const LoginButton = () => {
+interface LoginButtonProps {
+  tokenManager: TokenManager
+}
+
+export const LoginButton = (props: LoginButtonProps) => {
+  const { tokenManager: {isLoggedIn, logout} } = props
+  const history = useHistory()
+
+  if (isLoggedIn()) {
+    return <Button color="inherit" onClick={logout}>Logout</Button>
```

Comment:
i wonder about putting 'theme.palette.primary.main' or 'theme.palette.secondary.main' instead of inherit (which didn't work in another place). requires importing {theme} from ../../basestyle

Reply:
ah, thanks! you're right, we should set this explicitly -- i just did a bit of digging, and discovered that we don't actually need to pass in the theme object. Instead, we can just reference the value inside the theme that we're interested in (i guess in this case `secondary`, since the navBar background is `primary`).
```typescript
<Button color="secondary" onClick={logout}>Logout</Button>
```
^^ this currently works for me, since we're inside the theme provider context you built.

 In the case that we do need the theme object, we should probably access it via the theme provider, like this:
```typescript
import { useTheme } from '@mui/material/styles';
const theme = useTheme()
```

thanks again for pointing this out!